File: Practica1/Dice.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package practica1;
import java.util.Random;
/**
 *
 * @author manuel
 */

class Dice {
    private final float NHANGARSPROB;
    private final float NSHIELDSPROB;
    private final float NWEAPONSPROB;
    private final float FIRSTSHOTPROB;
    private Random generator;
    
    Dice (){
        
        NHANGARSPROB=0.25f;
        NSHIELDSPROB=0.25f;
        NWEAPONSPROB=0.33f;
        FIRSTSHOTPROB=0.5f;
        generator = new Random();
    }
    
    int initWithNHangars(){
        return ((generator.nextFloat() <= NHANGARSPROB )? 0:1);
    }
    
    int initWithNWeapons(){
        int valor = 1;
        float prob = generator.nextFloat();
        if (prob <= NWEAPONSPROB){
            valor = 1;
        }
        else if ( prob <= 2*NWEAPONSPROB ){
            valor = 2;
        }
        else if (prob <= 1f){
            valor = 3;
        }
            
        return valor;
    }
    
    int initWithNShields(){
        return ((generator.nextFloat() <= NSHIELDSPROB )? 0:1);
    }
    
    int whoStarts(int nPlayers){
       return generator.nextInt(nPlayers - 1 ); //revisar.
    }
    
    GameCharacter firstShot{
        
    }
    
    boolean spaceStationMoves(float speed){
        
    }
    
    
    
    
    
    
}

File: PRACTICA1/ruby/Dice.rb
module Deepspace

	class Dice
        def initialize 
			@NHANGARSPROB=0.25
			@NSHIELDSPROB=0.25
			@NWEAPONSPROB=0.33
			@FIRSTSHOTPROB=0.5
			@generator = Random.new()
        end 
        
        def initWithNHangars
			result = 0
			if @generator.rand() >= @NHANGARSPROB then
				result = 1
			end 
			return result
        
        end
        
        def initWithNWeapons
			valor = 1;
			prob = @generator.rand()
			if  prob <= @NWEAPONSPROB then
				valor = 1;
			elsif prob <= 2*@NWEAPONSPROB then
				valor = 2;
			elsif prob <= 1 then
				valor = 3;
			end
            
			return valor;
        
        end
        
        def initWithNShields
			result = 0
			if @generator.rand() >= @NSHIELDSPROB then
				result = 1
			end 
			return result
        
        end
        
        def whoStarts(nPlayers)
			@generator.rand(nPlayers)
        end 
        
        def firstShot 
			shot=0
			if @generator.rand() <= @FIRSTSHOTPROB then
				shot=GameCharacter::SPACESTATION
			else 
				shot=GameCharacter::ENEMYSTARSHIP
			end	
			
			return shot
        end
        
        def spaceStationMoves(speed)
			bool=false
			
			if @generator.rand() <= speed then
				bool=true
			end
			
			return bool
        end
        
        
	end
end 

